test/test_charm.py

```diff
@@ -32,47 +33,52 @@
 from .test_helpers import fake_script, fake_script_calls
 
 
-class TestCharm(unittest.TestCase):
+@pytest.fixture
+def tmpdir():
+    tmpdir = Path(tempfile.mkdtemp())
+    yield tmpdir
+    shutil.rmtree(str(tmpdir))
 
-    def setUp(self):
-        def restore_env(env: typing.Dict[str, str]):
-            os.environ.clear()
-            os.environ.update(env)
-        self.addCleanup(restore_env, os.environ.copy())
 
-        os.environ['PATH'] = os.pathsep.join([
-            str(Path(__file__).parent / 'bin'),
-            os.environ['PATH']])
-        os.environ['JUJU_UNIT_NAME'] = 'local/0'
+@pytest.fixture
```

Comment:
For what it's worth, pytest has [monkeypatch functionality specifically for environment variables](https://docs.pytest.org/en/6.2.x/monkeypatch.html#monkeypatching-environment-variables) so my guess (completely untested) would be that we could use that here.